### PR TITLE
Respond with a 413 when raising an `EntityTooLarge` 

### PR DESCRIPTION
**Pull request by danicheg**
Materializing my suggestions from https://github.com/http4s/http4s/pull/7602#issuecomment-2542072000

### CODE REVIEW

File: server/shared/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala [hunk 1]
```diff
@@ -21,11 +21,29 @@ package middleware
...
 import fs2._
+import org.http4s.headers.Connection
+import org.http4s.headers.`Content-Length`
 
 import scala.util.control.NoStackTrace
 
 object EntityLimiter {
-  final case class EntityTooLarge(limit: Long) extends Exception with NoStackTrace
+  final case class EntityTooLarge(limit: Long) extends Exception with NoStackTrace { self =>
+    def toMessageFailure: MessageFailure = new MessageFailure {
```
Comment (danicheg) [inline server/shared/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala:31]: In the 1.0 series, I'd like to address this more concisely by making `EntityTooLarge` straightforwardly extend `MessageFailure`. This change would eliminate the redundant conversion `EntityTooLarge` -> `MessageFailure` during error handling.

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala:31]: > In the 1.0 series, I'd like to address this more concisely by making `EntityTooLarge` straightforwardly extend `MessageFailure`.

Why can't we do this on the 0.23 series?

**Comment by danicheg** [comment]
I confusingly thought MiMa wouldn't allow us to do so, but no apparent issue was raised locally with the recent changes to make `EntityTooLarge` directly implement the `MessageFailure`. Thanks @armanbilge for raising that point.

**Review by armanbilge — APPROVED** [review]
Great change :)

**Review by samspills — APPROVED** [review]

**Pull request merged**